### PR TITLE
[DET-3044] fix tf.keras checkpoint export

### PR DESCRIPTION
**Pull request by sidneyw**
# Test Plan
- UI change:
  - [ ] add screenshots
  - [ ] React? build & check storybooks
- [ ] user-facing api change: modify documentation and examples
- [ ] user-facing api change: add the "User-facing API Change" label
- [ ] bug fix: add regression test
- [ ] bug fix: determine if there are other similar bugs in the codebase
- [ ] new feature: add test coverage for any user-facing aspects
- [ ] refactor: maintain existing code coverage

### CODE REVIEW

File: common/determined_common/experimental/checkpoint/_tf.py [hunk 1]
```diff
@@ -1,45 +1,52 @@
...
 
-    if not h5_paths and not saved_model_paths:
-        raise AssertionError(
-            "No checkpoint saved_model.pb or h5 files found at {}".format(ckpt_dir)
+def load_model(
+    ckpt_dir: pathlib.Path, metadata: Dict[str, Any], tags: Optional[List[str]] = None
+) -> AutoTrackable:
+    save_format = metadata.get("format", "")
+
+    if save_format == "saved_model" or "format" not in metadata:
```
Comment (aaron276h) [inline common/determined_common/experimental/checkpoint/_tf.py:16]: non-blocking: might improve readability if we do:

```
save_format = metadata.get("format", None)
if not save_format or cast(str, save_format) == "saved_model":
    ...
```

File: common/determined_common/experimental/checkpoint/_tf.py [hunk 2]
```diff
@@ -1,45 +1,52 @@
...
+            ckpt_dir.joinpath("code"),
+            config=metadata["experiment_config"],
+            hparams=metadata["hparams"],
         )
 
+        trial = cast(TFKerasTrial, trial)
+        model = trial.build_model()
+        model.load_weights(str(ckpt_dir.joinpath("determined-keras-model.h5")))
+        return model
+
```
Comment (aaron276h) [inline common/determined_common/experimental/checkpoint/_tf.py:30]: non-blocking: for completeness probably good to add a check here for unknown `save_format`

File: common/determined_common/experimental/checkpoint/_tf.py [hunk 3]
```diff
@@ -1,45 +1,52 @@
...
             )
+        )
 
-        if tags is None:
-            print('No tags specified. Loading "serve" tag from saved_model.')
-            tags = ["serve"]
-
-        saved_model_path = saved_model_paths[0]
-        return tf.compat.v1.saved_model.load_v2(str(saved_model_path.parent), tags)
+    if tags is None:
```
Comment (aaron276h) [inline common/determined_common/experimental/checkpoint/_tf.py:47]: non-blocking: a comment here would be useful

File: harness/determined/keras/_tf_keras_trial.py [hunk 4]
```diff
@@ -428,7 +428,9 @@ def _save_checkpoint(self, path: pathlib.Path) -> workload.Response:
         # Save model.
         self.model.save(path.joinpath("determined-keras-model.h5"), save_format="h5")
 
-        det.util.write_checkpoint_metadata(path, self.env, {"tensorflow_version": tf.__version__})
+        det.util.write_checkpoint_metadata(
+            path, self.env, {"tensorflow_version": tf.__version__, "format": "h5"}
```
[the remark this answers is not in the document]
Comment (sidneyw) [reply harness/determined/keras/_tf_keras_trial.py:432]: No we do not. I can add a version check to use `saved_model` in tf.2

File: common/determined_common/experimental/checkpoint/_checkpoint.py [hunk 5]
```diff
@@ -162,8 +162,18 @@ def load_from_path(path: str, tags: Optional[List[str]] = None, **kwargs: Any) -
                 <https://www.tensorflow.org/versions/r1.15/api_docs/python/tf/saved_model/load_v2>`_.
         """
         checkpoint_dir = pathlib.Path(path)
+        # We used MLflow's MLmodel checkpoint format in the past for
+        # serializing pytorch models.
```
Comment (yoavz) [inline common/determined_common/experimental/checkpoint/_checkpoint.py:166]: question: is this just for backwards compatability? if so, might be good to leave a note here that we intend to remove it in the future.

File: harness/determined/experimental/_native.py [hunk 6]
```diff
@@ -423,11 +425,44 @@ def local_execution_manager(new_directory: pathlib.Path) -> Iterator:
...
+        # entrypoint by changing the directory to model_def.
+        #
+        # Reference: https://docs.python.org/3/library/sys.html#sys.path
+        sys.path = [""] + sys.path
         yield
     finally:
         os.chdir(current_directory)
 
 
+def _local_trial_from_context(
```
Comment (yoavz) [inline harness/determined/experimental/_native.py:443]: blocking: we should re-use the `create_trial_instance` function here -- after the trial implementation is loaded from the context, most of the logic is the same.

File: harness/determined/experimental/_native.py [hunk 7]
```diff
@@ -423,11 +425,44 @@ def local_execution_manager(new_directory: pathlib.Path) -> Iterator:
...
+
+        # Python typically initializes sys.path[0] as the empty string when
+        # invoked interactively, which directs Python to search modules in the
+        # current directory first. However, this is _not_ happening when this
+        # Python function is invoked via the cli. We add it manually here so
+        # that _local_trial_from_context and test_one_batch can import the
+        # entrypoint by changing the directory to model_def.
+        #
+        # Reference: https://docs.python.org/3/library/sys.html#sys.path
+        sys.path = [""] + sys.path
```
Comment (yoavz) [inline harness/determined/experimental/_native.py:437]: question: why is this necessary to move to this path? my previous understanding was that we only needed to do this hack to alleviate the cli PATH startup problem.   Is `_local_trial_from_context` being called from any cli functions? 

 if we do need this functionality, `finally` should reset `sys.path` to what it was previously, or else we would have a nasty side effect with `load_trial` that modifies the `sys.path` whenever a user calls it.

Comment (sidneyw) [reply harness/determined/experimental/_native.py:437]: We need to add the directory to the `sys.path` in this case as well. Without this, the user code cannot be loaded as a module. I had to do something similar for the pytorch checkpoints as well. 

I can reset the path in the finally block.

File: harness/determined/keras/_tf_keras_trial.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (sidneyw) [reply harness/determined/keras/_tf_keras_trial.py:432]: We spoke offline. Using saved_model with tf.2 is non trivial. We are going to move forward with `h5` for both tf.1 and tf.2.

File: harness/determined/experimental/_native.py (hunk 7 above)
Comment (yoavz) [reply harness/determined/experimental/_native.py:441]: Sounds good.

File: common/determined_common/experimental/checkpoint/_tf.py [hunk 8]
```diff
@@ -1,45 +1,58 @@
...
             )
+        )
 
-        if tags is None:
-            print('No tags specified. Loading "serve" tag from saved_model.')
-            tags = ["serve"]
-
-        saved_model_path = saved_model_paths[0]
-        return tf.compat.v1.saved_model.load_v2(str(saved_model_path.parent), tags)
+    # Tensorflow uses tags to determined which metagraph to load. Most
```
Comment (aaron276h) [inline common/determined_common/experimental/checkpoint/_tf.py:49]: determined -> determine
